puppet/zulip_ops/templates/teleport_db.yaml.template.erb

```diff
@@ -0,0 +1,32 @@
+# See https://goteleport.com/docs/config-reference/ and
+# https://goteleport.com/docs/database-access/guides/postgres-self-hosted/
+teleport:
+  ca_pin: "sha256:df15ba56d56227e288ce183d7eee77a6bef552aaaa5dc25f0f5ea56494ce14c6"
+  auth_servers:
+    # Use the proxy address, to support running the db_service, which requires
+    # a reverse tunnel.
+    - teleport.zulipchat.net:443
+
+ssh_service:
+  enabled: no
+app_service:
+  enabled: no
+proxy_service:
+  enabled: no
+auth_service:
+  enabled: no
+
+db_service:
+  enabled: yes
+  databases:
+    - name: "<%= @hostname %>"
+      protocol: "postgres"
+      uri: "localhost:5432"
```

Comment:
This is IIRC the postgres port; so is the model that this listens on port 5432 and proxies to postgres?  If so, do we need to change postgres configuration to use a different port as well?

Reply:
No; postgres is unmodified; this line is telling the teleport database service where the postgres database lives, for it to connect to.  https://goteleport.com/docs/database-access/architecture/ describes this in more detail, but essentially:
 - Existing DB operations to the database are unchanged, modulo needing certs given by the teleport CA
 - The teleport database service sets up a tunnel back to the auth server, which advertises the DB connection
 - The auth server acts as a publicly-accessible border proxy, and proxies connections to the database through the tunnel to the `uri` provided, assuming they pass role-based access control.  The auth server will generate, on behalf of the user, the SSL cert needed to authenticate as the database user that the remote user requests.